Cargo.toml

```
@@ -4,7 +4,13 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
+axum = "0.8.3"
+futures = { version = "0.3.31", features = ["thread-pool"] }
+hyper = { version = "1.6.0", features = ["full"] } # for simulating test
+reqwest = "0.12.15"
+tokio = { version = "1.44.1", features = ["full"] }
+tower = "0.5.2"
 
 [lib]
 name = "newsletter_rs"
```

```suggestion
[lib]
name = "newsletter"
```
I suggest changing the name of the package too and remove the `-rs` suffix to it. You then don't need to specify the lib name here and let cargo infer its name from the package name.